modules/sql/server/vulnerability-assessment/main.bicep

```diff
@@ -17,12 +17,20 @@ param recurringScansEmailSubscriptionAdmins bool = false
 @description('Optional. Specifies an array of email addresses to which the scan notification is sent.')
 param recurringScansEmails array = []
 
-@description('Optional. A blob storage to hold the scan results.')
-param storageAccountResourceId string = ''
+@description('Required. A blob storage to hold the scan results.')
+param storageAccountResourceId string
+
+@description('Optional. Use Access Key to access the storage account. The storage account cannot be behind a firewall or virtual network. If an access key is not used, the SQL Server system assigned managed identity must be assigned the Storage Blob Data Contributor role on the storage account.')
+param useStorageAccountAccessKey bool = false
+
+@description('Optional. Create the Storage Blob Data Contributor role assignment on the storage account. Note, the role assignment must not already exist on the storage account.')
+param createStorageRoleAssignment bool = true
 
 @description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
 param enableDefaultTelemetry bool = true
 
+var splitStorageAccountResourceId = split(storageAccountResourceId, '/')
```

Comment:
Seems a bit odd to me if we keep this var around whole only using it in one location. Wouldn't it be easier to just have it's value in the location where it is used? I.e., last(split(storageAccountResourceId, '/'))

Reply:
don't mind doing that. We should then align the SQL MI one too. Agree? I can do it using this PR as well

Reply:
@AlexanderSehr updated both modules. SQL Serv ran successfully. SQL MI static tests passed (module dep in progress)